src/Client/Client.php

```diff
@@ -79,6 +95,12 @@ public function handleEvent($event, $name = null)
             }
         }
 
+        if (null !== $this->toSendLimit && $this->getToSend()->count() >= $this->toSendLimit) {
+            $this->send();
+
+            return;
```

Comment:
If `toSendLimit` is set and the limit is not reached, we still check `$immediateSend` at the end.
Am I wrong?